integration/integration.go

```diff
@@ -0,0 +1,184 @@
+	options := create.WithConfigFile(kindCfg)
+	ctx := cluster.NewContext(clusterName)
+	if err := ctx.Create(options); err != nil {
+		return "", err
+	}
+	defer func() {
+		ctx.Delete()
+	}()
+
+	clientset, err := getClientSet(ctx.KubeConfigPath())
+	if err != nil {
+		return "", err
+	}
+
+	jobYAML, err := ioutil.ReadFile(kubebenchYAML)
+	if err != nil {
+		return "", err
+	}
+
+	decoder := yaml.NewYAMLOrJSONDecoder(bytes.NewReader(jobYAML), len(jobYAML))
+	if err != nil {
+		return "", err
+	}
+
+	job := &batchv1.Job{}
+	if err := decoder.Decode(job); err != nil {
+		return "", err
+	}
+	job.Spec.Template.Spec.Containers[0].Image = kubebenchImg
+
+	if err := loadImageFromDocker(kubebenchImg, ctx); err != nil {
+		return "", err
+	}
+	
+
+	_, err = clientset.BatchV1().Jobs(apiv1.NamespaceDefault).Create(job)
+	if err != nil {
+		return "", err
+	}
+
+	clientset, err = getClientSet(ctx.KubeConfigPath())
+	if err != nil {
+		return "", err
+	}
+
+	p, err := findPodForJob(clientset, "kube-bench", timeout, ticker)
+	if err != nil {
+		return "", err
+	}
+
+	output := getPodLogs(clientset, p)
+	return output, nil
+}
+
+func getClientSet(configPath string) (*kubernetes.Clientset, error) {
+	config, err := clientcmd.BuildConfigFromFlags("", configPath)
+	if err != nil {
+		return nil, err
+	}
+	clientset, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		return nil, err
+	}
+
+	return clientset, nil
+}
+
+func int32Ptr(i int32) *int32 { return &i }
+
+func findPodForJob(clientset *kubernetes.Clientset, name string, tout, timer time.Duration) (*apiv1.Pod, error) {
+	timeout := time.After(tout)
+	failedPods := make(map[string]struct{})
+	for {
+	podfailed:
+		select {
+		case <-timeout:
+			return nil, fmt.Errorf("podList - time out: no Pod with %s", name)
+		default:
+			pods, err := clientset.CoreV1().Pods(apiv1.NamespaceDefault).List(metav1.ListOptions{})
+			if err != nil {
+				return nil, err
+			}
+			fmt.Printf("Found (%d) pods\n", len(pods.Items))
+			for _, cp := range pods.Items {
+				if _, found := failedPods[cp.Name]; found {
+					continue
+				}
+
+				if strings.HasPrefix(cp.Name, name) {
+					fmt.Printf("pod (%s) - %#v\n", cp.Name, cp.Status.Phase)
+					if cp.Status.Phase == apiv1.PodSucceeded {
+						return &cp, nil
+					}
+
+					if cp.Status.Phase == apiv1.PodFailed {
+						fmt.Printf("pod (%s) - %s - retrying...\n", cp.Name, cp.Status.Phase)
+						failedPods[cp.Name] = struct{}{}
+						break podfailed
+					}
+
+					// Pod still working
+					// Wait and try again...
+					ticker := time.NewTicker(timer)
+					for {
+						fmt.Println("using ticker and an timer...")
+						select {
+						case <-ticker.C:
+							thePod, err := clientset.CoreV1().Pods(apiv1.NamespaceDefault).Get(cp.Name, metav1.GetOptions{})
+							if err != nil {
+								return nil, err
+							}
+							fmt.Printf("thePod (%s) - status:%#v reason:%s message:%s\n", thePod.Name, thePod.Status.Phase, thePod.Status.Reason, thePod.Status.Message)
```

Comment:
The `reason` and `message` are always empty it seem when I ran it.

Reply:
Yeah I used to try to troubleshoot a pod in hanging mode. I'll remove it.